### PR TITLE
Split CI into discrete jobs

### LINKED ISSUE #890: Split tests and static analysis into separate actions in the CI
## Problem

Currently our "test" workflow is overloaded with static checks and unit tests and e2e tests. This makes them run one after the other meaning they take longer and also means that if an earlier one fails, you won't get any feedback about later checks.

## Description

Following Gutenberg's pattern, let's split the static analysis checks (linting, types) into a separate workflow entirely.

Additionally, create separate workflows for unit and e2e tests.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #890 by @sarayourfriend 

## Description

Adds a new `setup-node-env` composite action to make node and pnpm installation more DRY.

Splits unit tests, e2e test, storybook build, nuxt build, linting, i18n verification, and type checking into discrete jobs in the CI so that they all run at the same time.

## Testing Instructions

Check that the CI is running as expected compared to the behavior described in the yaml files.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
We moved the API repostory in the opposite direction, moving all separate jobs into a comprehensive CI + CD workflow that builds images, lints, runs tests, publishes images to GHCR and also deploys docs to GH pages. I believe that a comprehensive workflow is better than many separate ones because

- it's easier to share artifacts and environments
- you can visualise the entire process from the Actions diagram
- it's also concurrent (jobs that don't depend on each other will run in parallel even in the same workflow)

**Comment by sarayourfriend** [comment]
@dhruvkb Gotcha. I mostly just wanted to split each of the discrete tasks into their own jobs. I've pushed them into a single workflow now.

**Comment by sarayourfriend** [comment]
One disadvantage to putting everything into the same workflow is that you can't re-run discrete parts of it, you have to re-run every single check. I guess it doesn't cost anything to us really, it just always bugs me!

File: .lintstagedrc [hunk 1]
```diff
@@ -2,10 +2,7 @@
   "*.{ts,js,vue}": [
     "eslint --fix"
   ],
-  ".{ts,js,vue,json}": [
-    "prettier --write"
-  ],
-  "*.mdx": [
+  "*.{ts,js,vue,json,yml,yaml,mdx}": [
```
Comment (obulat) [inline .lintstagedrc:5]: Nice addition!

**Review by obulat — APPROVED** [review]

File: src/locales/scripts/json-to-pot.js [hunk 2]
```diff
@@ -2,8 +2,21 @@ const json = require('../en.json')
...
+  const existingPotFile = String(fs.readFileSync(fileName)).replace(
+    matchPotCreationDate,
+    ''
+  )
+  const potFile = createPotFile(json)
+
+  if (existingPotFile === potFile.replace(matchPotCreationDate, '')) {
+    console.log('No change detected in pot file, skipping write')
+    return process.exit(0)
+  }
```
Comment (zackkrida) [inline src/locales/scripts/json-to-pot.js:18]: very clever fix for easy diffing of the POT files.

Comment (dhruvkb) [reply src/locales/scripts/json-to-pot.js:18]: Wouldn't we need the replace on left side as well?

Comment (sarayourfriend) [reply src/locales/scripts/json-to-pot.js:18]: It's on line 9.

Comment (zackkrida) [reply src/locales/scripts/json-to-pot.js:18]: The date is stripped from the existing POT file in the preceding lines 9-12

Comment (dhruvkb) [reply src/locales/scripts/json-to-pot.js:18]: Ah cool, thanks. Don't know how I missed that.

**Review by zackkrida — APPROVED** [review]
LGTM. One non-blocking question, would there be any benefit to checking lint errors prior to types? I think it's probably even _better_ the way it is but i thought of the question and thought I'd pose it.

**Pull request merged**